File: templates/androidstarters-java/app/src/main/java/io/mvpstarter/sample/injection/component/ActivityComponent.java
package <%= appPackage %>.injection.component;

import dagger.Subcomponent;
import <%= appPackage %>.features.detail.DetailActivity;
import <%= appPackage %>.features.main.MainActivity;
import <%= appPackage %>.injection.PerActivity;
import <%= appPackage %>.injection.module.ActivityModule;

@PerActivity
@Subcomponent(modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(MainActivity mainActivity);

    void inject(DetailActivity detailActivity);
}

File: templates/androidstarters-kotlin/app/src/main/java/io/mvpstarter/sample/injection/component/ActivityComponent.kt
package <%= appPackage %>.injection.component

import <%= appPackage %>.injection.PerActivity
import <%= appPackage %>.injection.module.ActivityModule
import <%= appPackage %>.features.base.BaseActivity
import <%= appPackage %>.features.detail.DetailActivity
import <%= appPackage %>.features.main.MainActivity
import dagger.Subcomponent

@PerActivity
@Subcomponent(modules = arrayOf(ActivityModule::class))
interface ActivityComponent {
    fun inject(baseActivity: BaseActivity)

    fun inject(mainActivity: MainActivity)

    fun inject(detailActivity: DetailActivity)
}
